File: 628 - Maximum Product of Three Numbers.java
import java.util.Arrays;
import java.util.Collections;

class Solution {
    public int maximumProduct(int[] nums) {
        Arrays.sort(nums);
        int res = 0;
        if (nums[1] >= 0) {
            res = nums[nums.length - 1] * nums[nums.length - 2] * nums[nums.length - 3];
        } else {
            res = Math.max(nums[nums.length - 1] * nums[nums.length - 2] * nums[nums.length - 3], 
                            nums[0] * nums[1] * nums[nums.length - 1]);
        }
        return res;

    }
}

File: 628 - Maximum Product of Three Numbers.py
class Solution:
    def maximumProduct(self, nums: List[int]) -> int:
        nums.sort()
        if nums[1] >= 0:
            return nums.pop() * nums.pop() * nums.pop()
        else:
            return max(nums[0] * nums[1] * nums[-1], nums[-1] * nums[-2] * nums[-3])